presto-main/src/main/java/com/facebook/presto/execution/SqlTask.java

```diff
@@ -90,12 +93,21 @@ public static SqlTask createSqlTask(
             String nodeId,
             QueryContext queryContext,
             SqlTaskExecutionFactory sqlTaskExecutionFactory,
+            ExchangeClientManager exchangeClientManager,
```

Comment:
`ExchangeClientManager` is a task scope state. It doesn't have to be passed from the outside. I would create it inside.

Reply:
Not creating it inside is actually wrong, since there is one `ExchangeClientManager` holds for all the exchange clients and never removed.